"""
@generated by mypy-protobuf.  Do not edit manually!
isort:skip_file
"""
import builtins
import google.protobuf.descriptor
import google.protobuf.internal.enum_type_wrapper
import google.protobuf.message
import typing

DESCRIPTOR: google.protobuf.descriptor.FileDescriptor = ...

class DateRangeErrorEnum(google.protobuf.message.Message):
    DESCRIPTOR: google.protobuf.descriptor.Descriptor = ...
    class _DateRangeError(
        google.protobuf.internal.enum_type_wrapper._EnumTypeWrapper[DateRangeError.V],
        builtins.type,
    ):
        DESCRIPTOR: google.protobuf.descriptor.EnumDescriptor = ...
        UNSPECIFIED = DateRangeErrorEnum.DateRangeError.V(0)
        UNKNOWN = DateRangeErrorEnum.DateRangeError.V(1)
        INVALID_DATE = DateRangeErrorEnum.DateRangeError.V(2)
        START_DATE_AFTER_END_DATE = DateRangeErrorEnum.DateRangeError.V(3)
        CANNOT_SET_DATE_TO_PAST = DateRangeErrorEnum.DateRangeError.V(4)
        AFTER_MAXIMUM_ALLOWABLE_DATE = DateRangeErrorEnum.DateRangeError.V(5)
        CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED = (
            DateRangeErrorEnum.DateRangeError.V(6)
        )
    class DateRangeError(metaclass=_DateRangeError):
        V = typing.NewType("V", builtins.int)
    UNSPECIFIED = DateRangeErrorEnum.DateRangeError.V(0)
    UNKNOWN = DateRangeErrorEnum.DateRangeError.V(1)
    INVALID_DATE = DateRangeErrorEnum.DateRangeError.V(2)
    START_DATE_AFTER_END_DATE = DateRangeErrorEnum.DateRangeError.V(3)
    CANNOT_SET_DATE_TO_PAST = DateRangeErrorEnum.DateRangeError.V(4)
    AFTER_MAXIMUM_ALLOWABLE_DATE = DateRangeErrorEnum.DateRangeError.V(5)
    CANNOT_MODIFY_START_DATE_IF_ALREADY_STARTED = DateRangeErrorEnum.DateRangeError.V(6)
    def __init__(
        self,
    ) -> None: ...

global___DateRangeErrorEnum = DateRangeErrorEnum
